core/frontend/src/types/autopilot.ts

```diff
@@ -9,3 +9,31 @@ export enum Vehicle {
     Plane = 'Plane',
     Copter = 'Copter',
 }
+
+export enum EndpointType {
+    udpin = 'udpin',
+    udpout = 'udpout',
+    tcpin = 'tcpin',
+    tcpout = 'tcpout',
+    serial = 'serial',
+}
+
+export function userFriendlyEndpointType(type: EndpointType): string {
+  switch (type) {
+    case EndpointType.udpin: return 'UDP Server'
+    case EndpointType.udpout: return 'UDP Client'
+    case EndpointType.tcpin: return 'TCP Server'
+    case EndpointType.tcpout: return 'TCP Client'
+    case EndpointType.serial: return 'Serial'
+    default: return 'Undefined type'
+  }
+}
+export interface AutopilotEndpoint {
```

Comment:
isn't there a rule for empty lines between definitions ?

Reply:
Humn, apparently not.